contracts/SinglePlayerCommit.sol

```diff
@@ -51,213 +61,470 @@ contract SinglePlayerCommit is Ownable {
+        activities[_activityKey] = activity;
+        activityKeyList.push(_activityKey); 
+        emit ActivityUpdated(
+            activity.name, 
+            _activityKey, 
+            activity.oracle, 
+            activity.allowed, 
+            activity.exists);
         return _activityKey;
     }
 
-    function _changeCommitterBalance(uint256 amount, bool add) internal returns (bool) {
-        if (add) {
-            // increase committer's token balance
-            balances[msg.sender] = balances[msg.sender].add(amount);
-            // add to total committer balance sum
-            committerBalance = committerBalance.add(amount);
-        } else {
-            // decrease committer's token balance
-            balances[msg.sender] = balances[msg.sender].sub(amount);
-            // decrease total committer balance sum
-            committerBalance = committerBalance.sub(amount);
-        }
+    /// @notice Function to update oracle address of existing activity
+    /// @param _activityKey Keccak256 hashed, encoded name of activity
+    /// @param _oracleAddress Address of oracle for activity data    
+    /// @dev Check activity exists, update state, emit event
+    function updateActivityOracle(bytes32 _activityKey, address _oracleAddress) 
+        public
+        onlyOwner 
+        returns (bool success)
+    {
+        require(activities[_activityKey].exists, "SPC::_updateActivityOracle - activity does not exist");
+        Activity storage activity = activities[_activityKey];
+        activity.oracle = _oracleAddress;
+        emit ActivityUpdated(
+                activity.name, 
+                _activityKey, 
+                activity.oracle, 
+                activity.allowed, 
+                activity.exists
+            );
+        return true;
+    }
+
+    /// @notice Function to update availability of activity of existing activity
+    /// @param _activityKey Keccak256 hashed, encoded name of activity
+    /// @param _allowed Toggle for allowing new commitments with activity    
+    /// @dev Check activity exists, update state, emit event
+    function updateActivityAllowed(bytes32 _activityKey, bool _allowed) 
+        public
+        onlyOwner 
+        returns (bool success)
+    {
+        require(activities[_activityKey].exists, "SPC::_updateActivityOracle - activity does not exist");
+        Activity storage activity = activities[_activityKey];
+        activity.allowed = _allowed;
+        emit ActivityUpdated(
+                activity.name, 
+                _activityKey, 
+                activity.oracle, 
+                activity.allowed, 
+                activity.exists
+            );
+        return true;
+    }
 
+    /// @notice Function to 'delete' an existing activity. One way function, cannot be reversed.
+    /// @param _activityKey Keccak256 hashed, encoded name of activity
+    /// @dev Check activity exists, update state, emit event
+    function disableActivity(bytes32 _activityKey) 
```

Comment:
whats the difference between disabling and disallowing an activity as in `updateActivityAllowed`?

put another way, what is the benefit of deleting an activity vs. disallowing it?

Reply:
See other discussion